src/main/java/org/truffleruby/builtins/PrimitiveNodeConstructor.java

```diff
@@ -66,7 +65,7 @@ public RubyNode createCallPrimitiveNode(RubyContext context, Source source, Sour
 
     public RubyNode createInvokePrimitiveNode(RubyContext context, Source source, SourceIndexLength sourceSection, RubyNode[] arguments) {
         if (arguments.length != getPrimitiveArity()) {
-            throw new AssertionError("Incorrect number of arguments at " + RubyLanguage.fileLine(sourceSection.toSourceSection(source)));
+            throw new AssertionError("Incorrect number of arguments at " + context.getSourceLoader().fileLine(sourceSection.toSourceSection(source)));
```

Comment:
Outside the scope of this PR, but I hate when we throw `AssertionError`. It's just annoying to debug.

Reply:
RuntimeError or UnsupportedOperationException then?

Reply:
Yeah. Really, anything other than something that looks like it's an assertion.